src/repr/src/adt/regex.proto

```diff
@@ -12,5 +12,6 @@ syntax = "proto3";
 package mz_repr.adt.regex;
 
 message ProtoRegex {
-    string regex = 1;
+    string pattern = 1;
```

Comment:
Is name changing in a proto safe?

Reply:
Well, the linked thread says that "renaming fields is a wire compatible change". (The field name in the `ProtoRegex` struct changes, but I've of course changed the Rust code that is dealing with `ProtoRegex` directly.)

Also, we don't currently need backwards-compatibility for protobuf serialization, as far as I know.

Reply:
Sorry, I meant to say "I think so".